packages/core/utils/src/modules-sdk/medusa-internal-service.ts

```diff
@@ -304,7 +304,7 @@ export function MedusaInternalService<
       if (keySelectorForDataOnly.$or.length) {
         const entitiesToUpdate = await this.list(
           keySelectorForDataOnly,
-          {},
+          { take: null },
```

Comment:
cc. @adrien2p just so you are aware

Reply:
Can't we pass the selector to mikro orm instead of listing first?

Reply:
I thought we fixed that long time ago 😅